### PR TITLE
Add the ability to specify the disk the artisan command `lighthouse:print-schema` will write to via the `--disk` option

### PR DESCRIPTION
**Pull request by Carlwirkus**
- [x] Added or updated tests
- [x] Documented user facing changes
- [x] Updated CHANGELOG.md

**Changes**

- Add the ability to specify the disk the schema will print to via the `--disk` option OR the config option `lighthouse.print_schema_disk`

**Breaking changes**

### CODE REVIEW

**Comment by Carlwirkus** [comment]
Probably doesn't need the config option 🤷

File: src/lighthouse.php [hunk 1]
```diff
@@ -449,4 +449,14 @@
...
+
+    /*
+    |--------------------------------------------------------------------------
+    | Print Schema Disk
+    |--------------------------------------------------------------------------
+    |
+    | Define what disk should be used when printing the schema to file.
+    |
+    */
+    'print_schema_disk' => env('LIGHTHOUSE_PRINT_SCHEMA_DISK', 'local'),
```
Comment (spawnia) [inline src/lighthouse.php:461]: Not a fan of this option, just the CLI flag should suffice.

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,9 @@ You can find and compare releases at the [GitHub release page](https://github.co
 
 ## Unreleased
 
+### Added
+- Add the ability to specify the disk the schema will print too via the `--disk` option OR the config option `lighthouse.print_schema_disk`
```
Comment (spawnia) [inline CHANGELOG.md:13]:
```suggestion

- Add the ability to specify the disk the artisan command `lighthouse:print-schema` will write to via the `--disk` option
```

**Review by spawnia — CHANGES_REQUESTED** [review]
I am not convinced that writing to a disk is useful anyways. How about we allow specifying a path to write to instead?

File: src/lighthouse.php (hunk 1 above)
Comment (Carlwirkus) [reply src/lighthouse.php:461]: Sounds good

**Comment by spawnia** [comment]
Have you tested if this still selects the default disk if the option is not specified? What happens when a non-existant disk is specified?

**Comment by Carlwirkus** [comment]
> Have you tested if this still selects the default disk if the option is not specified?
Yes, it just selects the default disk.

> What happens when a non-existant disk is specified?
Not sure, I'll try tonight and see.

I see there's also issues with PHPStan, I'll see if I can get it setup locally and fix the conflicts. I've not worked with it before.

**Comment by Carlwirkus** [comment]
I've tested with the a invalid and missing disk. 
- With invalid disk you'll just get an error 'xyz' disk does not exist. 
- With no disk it will fallback to the default disk.

As for the errors with PHPStan, I'm not very experienced with using it. I've just cast the value to a string or null but let me know if there's a preferred way of dealing with this.

File: src/Console/PrintSchemaCommand.php [hunk 3]
```diff
@@ -52,8 +55,10 @@ public function handle(ASTCache $cache, Filesystem $storage, SchemaBuilder $sche
         }
 
         if ($this->option('write')) {
-            $storage->put($filename, $schemaString);
-            $this->info("Wrote schema to the default file storage (usually storage/app) as {$filename}.");
+            /** @var string|null $disk */
```
Comment (spawnia) [inline src/Console/PrintSchemaCommand.php:58]: This just ignores possible errors - PHPStan is correct that the value here might not be a string or null. Instead, do an actual check and throw if the value is anything else, indicating clearly to the user what was wrong.

Comment (Carlwirkus) [reply src/Console/PrintSchemaCommand.php]: I've added a check for a bool or array and returned an error to the console

**Pull request merged**